diff --git .formatter.exs .formatter.exs
index 6915976..4761678 100644
--- .formatter.exs
+++ .formatter.exs
@@ -1,4 +1,4 @@
 [
-  inputs: ["mix.exs", "config/*.exs"],
-  subdirectories: ["apps/*"]
+  import_deps: [:phoenix],
+  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
 ]
diff --git .gitignore .gitignore
index 12179ea..be002d1 100644
--- .gitignore
+++ .gitignore
@@ -18,3 +18,11 @@ erl_crash.dump
 
 # Also ignore archive artifacts (built via "mix archive.build").
 *.ez
+
+# Ignore package tarball (built via "mix hex.build").
+my_app-*.tar
+
+# Since we are building assets from assets/,
+# we ignore priv/static. You may want to comment
+# this depending on your deployment strategy.
+/priv/static/
diff --git README.md README.md
index 729adca..54e4276 100644
--- README.md
+++ README.md
@@ -1 +1,18 @@
-# MyApp.Umbrella
+# MyApp
+
+To start your Phoenix server:
+
+  * Install dependencies with `mix deps.get`
+  * Start Phoenix endpoint with `mix phx.server`
+
+Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+
+Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
+
+## Learn more
+
+  * Official website: https://www.phoenixframework.org/
+  * Guides: https://hexdocs.pm/phoenix/overview.html
+  * Docs: https://hexdocs.pm/phoenix
+  * Forum: https://elixirforum.com/c/phoenix-forum
+  * Source: https://github.com/phoenixframework/phoenix
diff --git apps/my_app/.formatter.exs b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/.formatter.exs
deleted file mode 100644
index 3d8ce11..0000000
--- apps/my_app/.formatter.exs
+++ /dev/null
@@ -1,3 +0,0 @@
-[
-  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
-]
diff --git apps/my_app/.gitignore b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/.gitignore
deleted file mode 100644
index 992da6d..0000000
--- apps/my_app/.gitignore
+++ /dev/null
@@ -1,23 +0,0 @@
-# The directory Mix will write compiled artifacts to.
-/_build/
-
-# If you run "mix test --cover", coverage assets end up here.
-/cover/
-
-# The directory Mix downloads your dependencies sources to.
-/deps/
-
-# Where 3rd-party dependencies like ExDoc output generated docs.
-/doc/
-
-# Ignore .fetch files in case you like to edit your project deps locally.
-/.fetch
-
-# If the VM crashes, it generates a dump, let's ignore it too.
-erl_crash.dump
-
-# Also ignore archive artifacts (built via "mix archive.build").
-*.ez
-
-# Ignore package tarball (built via "mix hex.build").
-my_app-*.tar
diff --git apps/my_app/README.md b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/README.md
deleted file mode 100644
index 4a0521a..0000000
--- apps/my_app/README.md
+++ /dev/null
@@ -1,3 +0,0 @@
-# MyApp
-
-**TODO: Add description**
diff --git apps/my_app/lib/my_app/application.ex b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/lib/my_app/application.ex
deleted file mode 100644
index d968a2f..0000000
--- apps/my_app/lib/my_app/application.ex
+++ /dev/null
@@ -1,18 +0,0 @@
-defmodule MyApp.Application do
-  # See https://hexdocs.pm/elixir/Application.html
-  # for more information on OTP Applications
-  @moduledoc false
-
-  use Application
-
-  def start(_type, _args) do
-    children = [
-      # Start the PubSub system
-      {Phoenix.PubSub, name: MyApp.PubSub}
-      # Start a worker by calling: MyApp.Worker.start_link(arg)
-      # {MyApp.Worker, arg}
-    ]
-
-    Supervisor.start_link(children, strategy: :one_for_one, name: MyApp.Supervisor)
-  end
-end
diff --git apps/my_app/mix.exs b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/mix.exs
deleted file mode 100644
index 07cf694..0000000
--- apps/my_app/mix.exs
+++ /dev/null
@@ -1,51 +0,0 @@
-defmodule MyApp.MixProject do
-  use Mix.Project
-
-  def project do
-    [
-      app: :my_app,
-      version: "0.1.0",
-      build_path: "../../_build",
-      config_path: "../../config/config.exs",
-      deps_path: "../../deps",
-      lockfile: "../../mix.lock",
-      elixir: "~> 1.7",
-      elixirc_paths: elixirc_paths(Mix.env()),
-      start_permanent: Mix.env() == :prod,
-      aliases: aliases(),
-      deps: deps()
-    ]
-  end
-
-  # Configuration for the OTP application.
-  #
-  # Type `mix help compile.app` for more information.
-  def application do
-    [
-      mod: {MyApp.Application, []},
-      extra_applications: [:logger, :runtime_tools]
-    ]
-  end
-
-  # Specifies which paths to compile per environment.
-  defp elixirc_paths(:test), do: ["lib", "test/support"]
-  defp elixirc_paths(_), do: ["lib"]
-
-  # Specifies your project dependencies.
-  #
-  # Type `mix help deps` for examples and options.
-  defp deps do
-    [
-      {:phoenix_pubsub, "~> 2.0"}
-    ]
-  end
-
-  # Aliases are shortcuts or tasks specific to the current project.
-  #
-  # See the documentation for `Mix` for more info on aliases.
-  defp aliases do
-    [
-      setup: ["deps.get"]
-    ]
-  end
-end
diff --git apps/my_app_web/.formatter.exs b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/.formatter.exs
deleted file mode 100644
index 4761678..0000000
--- apps/my_app_web/.formatter.exs
+++ /dev/null
@@ -1,4 +0,0 @@
-[
-  import_deps: [:phoenix],
-  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
-]
diff --git apps/my_app_web/.gitignore b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/.gitignore
deleted file mode 100644
index ea0dd82..0000000
--- apps/my_app_web/.gitignore
+++ /dev/null
@@ -1,28 +0,0 @@
-# The directory Mix will write compiled artifacts to.
-/_build/
-
-# If you run "mix test --cover", coverage assets end up here.
-/cover/
-
-# The directory Mix downloads your dependencies sources to.
-/deps/
-
-# Where 3rd-party dependencies like ExDoc output generated docs.
-/doc/
-
-# Ignore .fetch files in case you like to edit your project deps locally.
-/.fetch
-
-# If the VM crashes, it generates a dump, let's ignore it too.
-erl_crash.dump
-
-# Also ignore archive artifacts (built via "mix archive.build").
-*.ez
-
-# Ignore package tarball (built via "mix hex.build").
-my_app_web-*.tar
-
-# Since we are building assets from assets/,
-# we ignore priv/static. You may want to comment
-# this depending on your deployment strategy.
-/priv/static/
diff --git apps/my_app_web/README.md b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/README.md
deleted file mode 100644
index 4dcdde3..0000000
--- apps/my_app_web/README.md
+++ /dev/null
@@ -1,18 +0,0 @@
-# MyAppWeb
-
-To start your Phoenix server:
-
-  * Install dependencies with `mix deps.get`
-  * Start Phoenix endpoint with `mix phx.server`
-
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
-
-Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
-
-## Learn more
-
-  * Official website: https://www.phoenixframework.org/
-  * Guides: https://hexdocs.pm/phoenix/overview.html
-  * Docs: https://hexdocs.pm/phoenix
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
diff --git apps/my_app_web/lib/my_app_web/router.ex b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/router.ex
deleted file mode 100644
index 4abdff1..0000000
--- apps/my_app_web/lib/my_app_web/router.ex
+++ /dev/null
@@ -1,27 +0,0 @@
-defmodule MyAppWeb.Router do
-  use MyAppWeb, :router
-
-  pipeline :api do
-    plug :accepts, ["json"]
-  end
-
-  scope "/api", MyAppWeb do
-    pipe_through :api
-  end
-
-  # Enables LiveDashboard only for development
-  #
-  # If you want to use the LiveDashboard in production, you should put
-  # it behind authentication and allow only admins to access it.
-  # If your application does not have an admins-only section yet,
-  # you can use Plug.BasicAuth to set up some basic authentication
-  # as long as you are also using SSL (which you should anyway).
-  if Mix.env() in [:dev, :test] do
-    import Phoenix.LiveDashboard.Router
-
-    scope "/" do
-      pipe_through [:fetch_session, :protect_from_forgery]
-      live_dashboard "/dashboard", metrics: MyAppWeb.Telemetry
-    end
-  end
-end
diff --git apps/my_app_web/mix.exs b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/mix.exs
deleted file mode 100644
index 4da7d08..0000000
--- apps/my_app_web/mix.exs
+++ /dev/null
@@ -1,59 +0,0 @@
-defmodule MyAppWeb.MixProject do
-  use Mix.Project
-
-  def project do
-    [
-      app: :my_app_web,
-      version: "0.1.0",
-      build_path: "../../_build",
-      config_path: "../../config/config.exs",
-      deps_path: "../../deps",
-      lockfile: "../../mix.lock",
-      elixir: "~> 1.7",
-      elixirc_paths: elixirc_paths(Mix.env()),
-      compilers: [:phoenix, :gettext] ++ Mix.compilers(),
-      start_permanent: Mix.env() == :prod,
-      aliases: aliases(),
-      deps: deps()
-    ]
-  end
-
-  # Configuration for the OTP application.
-  #
-  # Type `mix help compile.app` for more information.
-  def application do
-    [
-      mod: {MyAppWeb.Application, []},
-      extra_applications: [:logger, :runtime_tools]
-    ]
-  end
-
-  # Specifies which paths to compile per environment.
-  defp elixirc_paths(:test), do: ["lib", "test/support"]
-  defp elixirc_paths(_), do: ["lib"]
-
-  # Specifies your project dependencies.
-  #
-  # Type `mix help deps` for examples and options.
-  defp deps do
-    [
-      {:phoenix, "~> 1.5.7"},
-      {:phoenix_live_dashboard, "~> 0.4"},
-      {:telemetry_metrics, "~> 0.4"},
-      {:telemetry_poller, "~> 0.4"},
-      {:gettext, "~> 0.11"},
-      {:my_app, in_umbrella: true},
-      {:jason, "~> 1.0"},
-      {:plug_cowboy, "~> 2.0"}
-    ]
-  end
-
-  # Aliases are shortcuts or tasks specific to the current project.
-  #
-  # See the documentation for `Mix` for more info on aliases.
-  defp aliases do
-    [
-      setup: ["deps.get"]
-    ]
-  end
-end
diff --git apps/my_app_web/test/test_helper.exs b/projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/test/test_helper.exs
deleted file mode 100644
index 869559e..0000000
--- apps/my_app_web/test/test_helper.exs
+++ /dev/null
@@ -1 +0,0 @@
-ExUnit.start()
diff --git config/config.exs config/config.exs
index f0c7234..b600774 100644
--- config/config.exs
+++ config/config.exs
@@ -1,21 +1,17 @@
-# This file is responsible for configuring your umbrella
-# and **all applications** and their dependencies with the
-# help of Mix.Config.
+# This file is responsible for configuring your application
+# and its dependencies with the aid of the Mix.Config module.
 #
-# Note that all applications in your umbrella share the
-# same configuration and dependencies, which is why they
-# all use the same configuration file. If you want different
-# configurations or dependencies per app, it is best to
-# move said applications out of the umbrella.
-use Mix.Config
-
+# This configuration file is loaded before any dependency and
+# is restricted to this project.
 
+# General application configuration
+use Mix.Config
 
-config :my_app_web,
-  generators: [context_app: :my_app]
+config :my_app,
+  generators: [binary_id: true]
 
 # Configures the endpoint
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   url: [host: "localhost"],
   secret_key_base: "[redacted]",
   render_errors: [view: MyAppWeb.ErrorView, accepts: ~w(json), layout: false],
diff --git config/dev.exs config/dev.exs
index 83402a1..a729e6b 100644
--- config/dev.exs
+++ config/dev.exs
@@ -6,7 +6,7 @@ use Mix.Config
 # The watchers configuration can be used to run external
 # watchers to your application. For example, we use it
 # with webpack to recompile .js and .css sources.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [port: 4000],
   debug_errors: true,
   code_reloader: true,
@@ -40,9 +40,9 @@ config :my_app_web, MyAppWeb.Endpoint,
 # Do not include metadata nor timestamps in development logs
 config :logger, :console, format: "[$level] $message\n"
 
-# Initialize plugs at runtime for faster development compilation
-config :phoenix, :plug_init_mode, :runtime
-
 # Set a higher stacktrace during development. Avoid configuring such
 # in production as building large stacktraces may be expensive.
 config :phoenix, :stacktrace_depth, 20
+
+# Initialize plugs at runtime for faster development compilation
+config :phoenix, :plug_init_mode, :runtime
diff --git config/prod.exs config/prod.exs
index 8310ab5..7443cb4 100644
--- config/prod.exs
+++ config/prod.exs
@@ -9,16 +9,19 @@ use Mix.Config
 # manifest is generated by the `mix phx.digest` task,
 # which you should run after static files are built and
 # before starting your production server.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   url: [host: "example.com", port: 80],
   cache_static_manifest: "priv/static/cache_manifest.json"
 
+# Do not print debug messages in production
+config :logger, level: :info
+
 # ## SSL Support
 #
 # To get SSL working, you will need to add the `https` key
 # to the previous section and set your `:url` port to 443:
 #
-#     config :my_app_web, MyAppWeb.Endpoint,
+#     config :my_app, MyAppWeb.Endpoint,
 #       ...
 #       url: [host: "example.com", port: 443],
 #       https: [
@@ -42,14 +45,11 @@ config :my_app_web, MyAppWeb.Endpoint,
 # We also recommend setting `force_ssl` in your endpoint, ensuring
 # no data is ever sent via http, always redirecting to https:
 #
-#     config :my_app_web, MyAppWeb.Endpoint,
+#     config :my_app, MyAppWeb.Endpoint,
 #       force_ssl: [hsts: true]
 #
 # Check `Plug.SSL` for all available options in `force_ssl`.
 
-# Do not print debug messages in production
-config :logger, level: :info
-
 # Finally import the config/prod.secret.exs which loads secrets
 # and configuration from environment variables.
 import_config "prod.secret.exs"
diff --git config/prod.secret.exs config/prod.secret.exs
index 702c128..3568fa4 100644
--- config/prod.secret.exs
+++ config/prod.secret.exs
@@ -11,7 +11,7 @@ secret_key_base =
     You can generate one by calling: mix phx.gen.secret
     """
 
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [
     port: String.to_integer(System.get_env("PORT") || "4000"),
     transport_options: [socket_opts: [:inet6]]
@@ -23,7 +23,7 @@ config :my_app_web, MyAppWeb.Endpoint,
 # If you are doing OTP releases, you need to instruct Phoenix
 # to start each relevant endpoint:
 #
-#     config :my_app_web, MyAppWeb.Endpoint, server: true
+#     config :my_app, MyAppWeb.Endpoint, server: true
 #
 # Then you can assemble a release by calling `mix release`.
 # See `mix help release` for more information.
diff --git config/test.exs config/test.exs
index 8fd529b..d363390 100644
--- config/test.exs
+++ config/test.exs
@@ -2,7 +2,7 @@ use Mix.Config
 
 # We don't run a server during test. If one is required,
 # you can enable the server option below.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [port: 4002],
   server: false
 
diff --git apps/my_app_web/lib/my_app_web/application.ex lib/my_app/application.ex
similarity index 72%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/application.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app/application.ex
index fbdce53..9ecbdae 100644
--- apps/my_app_web/lib/my_app_web/application.ex
+++ lib/my_app/application.ex
@@ -1,4 +1,4 @@
-defmodule MyAppWeb.Application do
+defmodule MyApp.Application do
   # See https://hexdocs.pm/elixir/Application.html
   # for more information on OTP Applications
   @moduledoc false
@@ -9,15 +9,17 @@ defmodule MyAppWeb.Application do
     children = [
       # Start the Telemetry supervisor
       MyAppWeb.Telemetry,
+      # Start the PubSub system
+      {Phoenix.PubSub, name: MyApp.PubSub},
       # Start the Endpoint (http/https)
       MyAppWeb.Endpoint
-      # Start a worker by calling: MyAppWeb.Worker.start_link(arg)
-      # {MyAppWeb.Worker, arg}
+      # Start a worker by calling: MyApp.Worker.start_link(arg)
+      # {MyApp.Worker, arg}
     ]
 
     # See https://hexdocs.pm/elixir/Supervisor.html
     # for other strategies and supported options
-    opts = [strategy: :one_for_one, name: MyAppWeb.Supervisor]
+    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
     Supervisor.start_link(children, opts)
   end
 
diff --git apps/my_app/lib/my_app.ex lib/my_app.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/lib/my_app.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app.ex
diff --git apps/my_app_web/lib/my_app_web/channels/user_socket.ex lib/my_app_web/channels/user_socket.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/channels/user_socket.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/channels/user_socket.ex
diff --git apps/my_app_web/lib/my_app_web/endpoint.ex lib/my_app_web/endpoint.ex
similarity index 76%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/endpoint.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/endpoint.ex
index d5bef21..85a4675 100644
--- apps/my_app_web/lib/my_app_web/endpoint.ex
+++ lib/my_app_web/endpoint.ex
@@ -1,28 +1,26 @@
 defmodule MyAppWeb.Endpoint do
-  use Phoenix.Endpoint, otp_app: :my_app_web
+  use Phoenix.Endpoint, otp_app: :my_app
 
   # The session will be stored in the cookie and signed,
   # this means its contents can be read but not tampered with.
   # Set :encryption_salt if you would also like to encrypt it.
   @session_options [
     store: :cookie,
-    key: "_my_app_web_key",
-    signing_salt: "1ThPJBR5"
+    key: "_my_app_key",
+    signing_salt: "[redacted]"
   ]
 
   socket "/socket", MyAppWeb.UserSocket,
     websocket: true,
     longpoll: false
 
-  socket "/live", Phoenix.LiveView.Socket, websocket: [connect_info: [session: @session_options]]
-
   # Serve at "/" the static files from "priv/static" directory.
   #
   # You should set gzip to true if you are running phx.digest
   # when deploying your static files in production.
   plug Plug.Static,
     at: "/",
-    from: :my_app_web,
+    from: :my_app,
     gzip: false,
     only: ~w(css fonts images js favicon.ico robots.txt)
 
@@ -32,10 +30,6 @@ defmodule MyAppWeb.Endpoint do
     plug Phoenix.CodeReloader
   end
 
-  plug Phoenix.LiveDashboard.RequestLogger,
-    param_key: "request_logger",
-    cookie_key: "request_logger"
-
   plug Plug.RequestId
   plug Plug.Telemetry, event_prefix: [:phoenix, :endpoint]
 
diff --git apps/my_app_web/lib/my_app_web/gettext.ex lib/my_app_web/gettext.ex
similarity index 94%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/gettext.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/gettext.ex
index 9c6644d..7919b97 100644
--- apps/my_app_web/lib/my_app_web/gettext.ex
+++ lib/my_app_web/gettext.ex
@@ -20,5 +20,5 @@ defmodule MyAppWeb.Gettext do
 
   See the [Gettext Docs](https://hexdocs.pm/gettext) for detailed usage.
   """
-  use Gettext, otp_app: :my_app_web
+  use Gettext, otp_app: :my_app
 end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/router.ex lib/my_app_web/router.ex
new file mode 100644
index 0000000..5f5f6fc
--- /dev/null
+++ lib/my_app_web/router.ex
@@ -0,0 +1,11 @@
+defmodule MyAppWeb.Router do
+  use MyAppWeb, :router
+
+  pipeline :api do
+    plug :accepts, ["json"]
+  end
+
+  scope "/api", MyAppWeb do
+    pipe_through :api
+  end
+end
diff --git apps/my_app_web/lib/my_app_web/telemetry.ex lib/my_app_web/telemetry.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/telemetry.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/telemetry.ex
diff --git apps/my_app_web/lib/my_app_web/views/error_helpers.ex lib/my_app_web/views/error_helpers.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/views/error_helpers.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/views/error_helpers.ex
diff --git apps/my_app_web/lib/my_app_web/views/error_view.ex lib/my_app_web/views/error_view.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web/views/error_view.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web/views/error_view.ex
diff --git apps/my_app_web/lib/my_app_web.ex lib/my_app_web.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/lib/my_app_web.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/lib/my_app_web.ex
diff --git mix.exs mix.exs
index 099c509..b938d68 100644
--- mix.exs
+++ mix.exs
@@ -1,30 +1,45 @@
-defmodule MyApp.Umbrella.MixProject do
+defmodule MyApp.MixProject do
   use Mix.Project
 
   def project do
     [
-      apps_path: "apps",
+      app: :my_app,
       version: "0.1.0",
+      elixir: "~> 1.7",
+      elixirc_paths: elixirc_paths(Mix.env()),
+      compilers: [:phoenix, :gettext] ++ Mix.compilers(),
       start_permanent: Mix.env() == :prod,
-      deps: deps(),
-      aliases: aliases()
+      aliases: aliases(),
+      deps: deps()
     ]
   end
 
-  # Dependencies can be Hex packages:
+  # Configuration for the OTP application.
   #
-  #   {:mydep, "~> 0.3.0"}
-  #
-  # Or git/path repositories:
-  #
-  #   {:mydep, git: "https://github.com/elixir-lang/mydep.git", tag: "0.1.0"}
-  #
-  # Type "mix help deps" for more examples and options.
+  # Type `mix help compile.app` for more information.
+  def application do
+    [
+      mod: {MyApp.Application, []},
+      extra_applications: [:logger, :runtime_tools]
+    ]
+  end
+
+  # Specifies which paths to compile per environment.
+  defp elixirc_paths(:test), do: ["lib", "test/support"]
+  defp elixirc_paths(_), do: ["lib"]
+
+  # Specifies your project dependencies.
   #
-  # Dependencies listed here are available only for this project
-  # and cannot be accessed from applications inside the apps/ folder.
+  # Type `mix help deps` for examples and options.
   defp deps do
-    []
+    [
+      {:phoenix, "~> 1.5.7"},
+      {:telemetry_metrics, "~> 0.4"},
+      {:telemetry_poller, "~> 0.4"},
+      {:gettext, "~> 0.11"},
+      {:jason, "~> 1.0"},
+      {:plug_cowboy, "~> 2.0"}
+    ]
   end
 
   # Aliases are shortcuts or tasks specific to the current project.
@@ -33,13 +48,9 @@ defmodule MyApp.Umbrella.MixProject do
   #     $ mix setup
   #
   # See the documentation for `Mix` for more info on aliases.
-  #
-  # Aliases listed here are available only for this project
-  # and cannot be accessed from applications inside the apps/ folder.
   defp aliases do
     [
-      # run `mix setup` in all child apps
-      setup: ["cmd mix setup"]
+      setup: ["deps.get"]
     ]
   end
 end
diff --git apps/my_app_web/priv/gettext/en/LC_MESSAGES/errors.po priv/gettext/en/LC_MESSAGES/errors.po
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/priv/gettext/en/LC_MESSAGES/errors.po
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/priv/gettext/en/LC_MESSAGES/errors.po
diff --git apps/my_app_web/priv/gettext/errors.pot priv/gettext/errors.pot
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/priv/gettext/errors.pot
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/priv/gettext/errors.pot
diff --git apps/my_app_web/test/my_app_web/views/error_view_test.exs test/my_app_web/views/error_view_test.exs
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/test/my_app_web/views/error_view_test.exs
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/test/my_app_web/views/error_view_test.exs
diff --git apps/my_app_web/test/support/channel_case.ex test/support/channel_case.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/test/support/channel_case.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/test/support/channel_case.ex
diff --git apps/my_app_web/test/support/conn_case.ex test/support/conn_case.ex
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app_web/test/support/conn_case.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/test/support/conn_case.ex
diff --git apps/my_app/test/test_helper.exs test/test_helper.exs
similarity index 100%
rename from projects/phoenix/1.5.7/database-noecto-nohtml-nowebpack-umbrella/apps/my_app/test/test_helper.exs
rename to projects/phoenix/1.5.7/binaryid-database-noecto-nohtml-nowebpack-nodash/test/test_helper.exs
