diff --git .formatter.exs .formatter.exs
index 6915976..4761678 100644
--- .formatter.exs
+++ .formatter.exs
@@ -1,4 +1,4 @@
 [
-  inputs: ["mix.exs", "config/*.exs"],
-  subdirectories: ["apps/*"]
+  import_deps: [:phoenix],
+  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
 ]
diff --git .gitignore .gitignore
index 12179ea..8b3a72e 100644
--- .gitignore
+++ .gitignore
@@ -18,3 +18,17 @@ erl_crash.dump
 
 # Also ignore archive artifacts (built via "mix archive.build").
 *.ez
+
+# Ignore package tarball (built via "mix hex.build").
+my_app-*.tar
+
+# If NPM crashes, it generates a log, let's ignore it too.
+npm-debug.log
+
+# The directory NPM downloads your dependencies sources to.
+/assets/node_modules/
+
+# Since we are building assets from assets/,
+# we ignore priv/static. You may want to comment
+# this depending on your deployment strategy.
+/priv/static/
diff --git README.md README.md
index 729adca..0d08a74 100644
--- README.md
+++ README.md
@@ -1 +1,19 @@
-# MyApp.Umbrella
+# MyApp
+
+To start your Phoenix server:
+
+  * Install dependencies with `mix deps.get`
+  * Install Node.js dependencies with `npm install` inside the `assets` directory
+  * Start Phoenix endpoint with `mix phx.server`
+
+Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+
+Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
+
+## Learn more
+
+  * Official website: https://www.phoenixframework.org/
+  * Guides: https://hexdocs.pm/phoenix/overview.html
+  * Docs: https://hexdocs.pm/phoenix
+  * Forum: https://elixirforum.com/c/phoenix-forum
+  * Source: https://github.com/phoenixframework/phoenix
diff --git apps/my_app/.formatter.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/.formatter.exs
deleted file mode 100644
index 3d8ce11..0000000
--- apps/my_app/.formatter.exs
+++ /dev/null
@@ -1,3 +0,0 @@
-[
-  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
-]
diff --git apps/my_app/.gitignore b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/.gitignore
deleted file mode 100644
index 992da6d..0000000
--- apps/my_app/.gitignore
+++ /dev/null
@@ -1,23 +0,0 @@
-# The directory Mix will write compiled artifacts to.
-/_build/
-
-# If you run "mix test --cover", coverage assets end up here.
-/cover/
-
-# The directory Mix downloads your dependencies sources to.
-/deps/
-
-# Where 3rd-party dependencies like ExDoc output generated docs.
-/doc/
-
-# Ignore .fetch files in case you like to edit your project deps locally.
-/.fetch
-
-# If the VM crashes, it generates a dump, let's ignore it too.
-erl_crash.dump
-
-# Also ignore archive artifacts (built via "mix archive.build").
-*.ez
-
-# Ignore package tarball (built via "mix hex.build").
-my_app-*.tar
diff --git apps/my_app/README.md b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/README.md
deleted file mode 100644
index 4a0521a..0000000
--- apps/my_app/README.md
+++ /dev/null
@@ -1,3 +0,0 @@
-# MyApp
-
-**TODO: Add description**
diff --git apps/my_app/lib/my_app/application.ex b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/lib/my_app/application.ex
deleted file mode 100644
index d968a2f..0000000
--- apps/my_app/lib/my_app/application.ex
+++ /dev/null
@@ -1,18 +0,0 @@
-defmodule MyApp.Application do
-  # See https://hexdocs.pm/elixir/Application.html
-  # for more information on OTP Applications
-  @moduledoc false
-
-  use Application
-
-  def start(_type, _args) do
-    children = [
-      # Start the PubSub system
-      {Phoenix.PubSub, name: MyApp.PubSub}
-      # Start a worker by calling: MyApp.Worker.start_link(arg)
-      # {MyApp.Worker, arg}
-    ]
-
-    Supervisor.start_link(children, strategy: :one_for_one, name: MyApp.Supervisor)
-  end
-end
diff --git apps/my_app/mix.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/mix.exs
deleted file mode 100644
index 07cf694..0000000
--- apps/my_app/mix.exs
+++ /dev/null
@@ -1,51 +0,0 @@
-defmodule MyApp.MixProject do
-  use Mix.Project
-
-  def project do
-    [
-      app: :my_app,
-      version: "0.1.0",
-      build_path: "../../_build",
-      config_path: "../../config/config.exs",
-      deps_path: "../../deps",
-      lockfile: "../../mix.lock",
-      elixir: "~> 1.7",
-      elixirc_paths: elixirc_paths(Mix.env()),
-      start_permanent: Mix.env() == :prod,
-      aliases: aliases(),
-      deps: deps()
-    ]
-  end
-
-  # Configuration for the OTP application.
-  #
-  # Type `mix help compile.app` for more information.
-  def application do
-    [
-      mod: {MyApp.Application, []},
-      extra_applications: [:logger, :runtime_tools]
-    ]
-  end
-
-  # Specifies which paths to compile per environment.
-  defp elixirc_paths(:test), do: ["lib", "test/support"]
-  defp elixirc_paths(_), do: ["lib"]
-
-  # Specifies your project dependencies.
-  #
-  # Type `mix help deps` for examples and options.
-  defp deps do
-    [
-      {:phoenix_pubsub, "~> 2.0"}
-    ]
-  end
-
-  # Aliases are shortcuts or tasks specific to the current project.
-  #
-  # See the documentation for `Mix` for more info on aliases.
-  defp aliases do
-    [
-      setup: ["deps.get"]
-    ]
-  end
-end
diff --git apps/my_app_web/.formatter.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/.formatter.exs
deleted file mode 100644
index 4761678..0000000
--- apps/my_app_web/.formatter.exs
+++ /dev/null
@@ -1,4 +0,0 @@
-[
-  import_deps: [:phoenix],
-  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
-]
diff --git apps/my_app_web/.gitignore b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/.gitignore
deleted file mode 100644
index 5f1d967..0000000
--- apps/my_app_web/.gitignore
+++ /dev/null
@@ -1,34 +0,0 @@
-# The directory Mix will write compiled artifacts to.
-/_build/
-
-# If you run "mix test --cover", coverage assets end up here.
-/cover/
-
-# The directory Mix downloads your dependencies sources to.
-/deps/
-
-# Where 3rd-party dependencies like ExDoc output generated docs.
-/doc/
-
-# Ignore .fetch files in case you like to edit your project deps locally.
-/.fetch
-
-# If the VM crashes, it generates a dump, let's ignore it too.
-erl_crash.dump
-
-# Also ignore archive artifacts (built via "mix archive.build").
-*.ez
-
-# Ignore package tarball (built via "mix hex.build").
-my_app_web-*.tar
-
-# If NPM crashes, it generates a log, let's ignore it too.
-npm-debug.log
-
-# The directory NPM downloads your dependencies sources to.
-/assets/node_modules/
-
-# Since we are building assets from assets/,
-# we ignore priv/static. You may want to comment
-# this depending on your deployment strategy.
-/priv/static/
diff --git apps/my_app_web/README.md b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/README.md
deleted file mode 100644
index 6d4b014..0000000
--- apps/my_app_web/README.md
+++ /dev/null
@@ -1,19 +0,0 @@
-# MyAppWeb
-
-To start your Phoenix server:
-
-  * Install dependencies with `mix deps.get`
-  * Install Node.js dependencies with `npm install` inside the `assets` directory
-  * Start Phoenix endpoint with `mix phx.server`
-
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
-
-Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
-
-## Learn more
-
-  * Official website: https://www.phoenixframework.org/
-  * Guides: https://hexdocs.pm/phoenix/overview.html
-  * Docs: https://hexdocs.pm/phoenix
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
diff --git apps/my_app_web/assets/css/app.scss b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/css/app.scss
deleted file mode 100644
index 5c2c9b5..0000000
--- apps/my_app_web/assets/css/app.scss
+++ /dev/null
@@ -1,36 +0,0 @@
-/* This file is for your main application css. */
-@import "./phoenix.css";
-
-/* Alerts and form errors */
-.alert {
-  padding: 15px;
-  margin-bottom: 20px;
-  border: 1px solid transparent;
-  border-radius: 4px;
-}
-.alert-info {
-  color: #31708f;
-  background-color: #d9edf7;
-  border-color: #bce8f1;
-}
-.alert-warning {
-  color: #8a6d3b;
-  background-color: #fcf8e3;
-  border-color: #faebcc;
-}
-.alert-danger {
-  color: #a94442;
-  background-color: #f2dede;
-  border-color: #ebccd1;
-}
-.alert p {
-  margin-bottom: 0;
-}
-.alert:empty {
-  display: none;
-}
-.invalid-feedback {
-  color: #a94442;
-  display: block;
-  margin: -1rem 0 2rem;
-}
diff --git apps/my_app_web/assets/js/app.js b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/js/app.js
deleted file mode 100644
index 5f6ee98..0000000
--- apps/my_app_web/assets/js/app.js
+++ /dev/null
@@ -1,15 +0,0 @@
-// We need to import the CSS so that webpack will load it.
-// The MiniCssExtractPlugin is used to separate it out into
-// its own CSS file.
-import "../css/app.scss"
-
-// webpack automatically bundles all modules in your
-// entry points. Those entry points can be configured
-// in "webpack.config.js".
-//
-// Import deps with the dep name or local files with a relative path, for example:
-//
-//     import {Socket} from "phoenix"
-//     import socket from "./socket"
-//
-
diff --git apps/my_app_web/assets/js/socket.js b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/js/socket.js
deleted file mode 100644
index 09929ab..0000000
--- apps/my_app_web/assets/js/socket.js
+++ /dev/null
@@ -1,63 +0,0 @@
-// NOTE: The contents of this file will only be executed if
-// you uncomment its entry in "assets/js/app.js".
-
-// To use Phoenix channels, the first step is to import Socket,
-// and connect at the socket path in "lib/web/endpoint.ex".
-//
-// Pass the token on params as below. Or remove it
-// from the params if you are not using authentication.
-import {Socket} from "phoenix"
-
-let socket = new Socket("/socket", {params: {token: window.userToken}})
-
-// When you connect, you'll often need to authenticate the client.
-// For example, imagine you have an authentication plug, `MyAuth`,
-// which authenticates the session and assigns a `:current_user`.
-// If the current user exists you can assign the user's token in
-// the connection for use in the layout.
-//
-// In your "lib/web/router.ex":
-//
-//     pipeline :browser do
-//       ...
-//       plug MyAuth
-//       plug :put_user_token
-//     end
-//
-//     defp put_user_token(conn, _) do
-//       if current_user = conn.assigns[:current_user] do
-//         token = Phoenix.Token.sign(conn, "user socket", current_user.id)
-//         assign(conn, :user_token, token)
-//       else
-//         conn
-//       end
-//     end
-//
-// Now you need to pass this token to JavaScript. You can do so
-// inside a script tag in "lib/web/templates/layout/app.html.eex":
-//
-//     <script>window.userToken = "<%= assigns[:user_token] %>";</script>
-//
-// You will need to verify the user token in the "connect/3" function
-// in "lib/web/channels/user_socket.ex":
-//
-//     def connect(%{"token" => token}, socket, _connect_info) do
-//       # max_age: 1209600 is equivalent to two weeks in seconds
-//       case Phoenix.Token.verify(socket, "user socket", token, max_age: 1209600) do
-//         {:ok, user_id} ->
-//           {:ok, assign(socket, :user, user_id)}
-//         {:error, reason} ->
-//           :error
-//       end
-//     end
-//
-// Finally, connect to the socket:
-socket.connect()
-
-// Now that you are connected, you can join channels with a topic:
-let channel = socket.channel("topic:subtopic", {})
-channel.join()
-  .receive("ok", resp => { console.log("Joined successfully", resp) })
-  .receive("error", resp => { console.log("Unable to join", resp) })
-
-export default socket
diff --git apps/my_app_web/lib/my_app_web/router.ex b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/router.ex
deleted file mode 100644
index 5f5f6fc..0000000
--- apps/my_app_web/lib/my_app_web/router.ex
+++ /dev/null
@@ -1,11 +0,0 @@
-defmodule MyAppWeb.Router do
-  use MyAppWeb, :router
-
-  pipeline :api do
-    plug :accepts, ["json"]
-  end
-
-  scope "/api", MyAppWeb do
-    pipe_through :api
-  end
-end
diff --git apps/my_app_web/mix.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/mix.exs
deleted file mode 100644
index 4e4a71f..0000000
--- apps/my_app_web/mix.exs
+++ /dev/null
@@ -1,58 +0,0 @@
-defmodule MyAppWeb.MixProject do
-  use Mix.Project
-
-  def project do
-    [
-      app: :my_app_web,
-      version: "0.1.0",
-      build_path: "../../_build",
-      config_path: "../../config/config.exs",
-      deps_path: "../../deps",
-      lockfile: "../../mix.lock",
-      elixir: "~> 1.7",
-      elixirc_paths: elixirc_paths(Mix.env()),
-      compilers: [:phoenix, :gettext] ++ Mix.compilers(),
-      start_permanent: Mix.env() == :prod,
-      aliases: aliases(),
-      deps: deps()
-    ]
-  end
-
-  # Configuration for the OTP application.
-  #
-  # Type `mix help compile.app` for more information.
-  def application do
-    [
-      mod: {MyAppWeb.Application, []},
-      extra_applications: [:logger, :runtime_tools]
-    ]
-  end
-
-  # Specifies which paths to compile per environment.
-  defp elixirc_paths(:test), do: ["lib", "test/support"]
-  defp elixirc_paths(_), do: ["lib"]
-
-  # Specifies your project dependencies.
-  #
-  # Type `mix help deps` for examples and options.
-  defp deps do
-    [
-      {:phoenix, "~> 1.5.7"},
-      {:telemetry_metrics, "~> 0.4"},
-      {:telemetry_poller, "~> 0.4"},
-      {:gettext, "~> 0.11"},
-      {:my_app, in_umbrella: true},
-      {:jason, "~> 1.0"},
-      {:plug_cowboy, "~> 2.0"}
-    ]
-  end
-
-  # Aliases are shortcuts or tasks specific to the current project.
-  #
-  # See the documentation for `Mix` for more info on aliases.
-  defp aliases do
-    [
-      setup: ["deps.get", "cmd npm install --prefix assets"]
-    ]
-  end
-end
diff --git apps/my_app_web/test/my_app_web/views/error_view_test.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/test/my_app_web/views/error_view_test.exs
deleted file mode 100644
index 05a065b..0000000
--- apps/my_app_web/test/my_app_web/views/error_view_test.exs
+++ /dev/null
@@ -1,15 +0,0 @@
-defmodule MyAppWeb.ErrorViewTest do
-  use MyAppWeb.ConnCase, async: true
-
-  # Bring render/3 and render_to_string/3 for testing custom views
-  import Phoenix.View
-
-  test "renders 404.json" do
-    assert render(MyAppWeb.ErrorView, "404.json", []) == %{errors: %{detail: "Not Found"}}
-  end
-
-  test "renders 500.json" do
-    assert render(MyAppWeb.ErrorView, "500.json", []) ==
-             %{errors: %{detail: "Internal Server Error"}}
-  end
-end
diff --git apps/my_app_web/test/test_helper.exs b/projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/test/test_helper.exs
deleted file mode 100644
index 869559e..0000000
--- apps/my_app_web/test/test_helper.exs
+++ /dev/null
@@ -1 +0,0 @@
-ExUnit.start()
diff --git apps/my_app_web/assets/.babelrc assets/.babelrc
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/.babelrc
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/.babelrc
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/css/app.scss assets/css/app.scss
new file mode 100644
index 0000000..f7775db
--- /dev/null
+++ assets/css/app.scss
@@ -0,0 +1,91 @@
+/* This file is for your main application css. */
+@import "./phoenix.css";
+@import "../node_modules/nprogress/nprogress.css";
+
+/* LiveView specific classes for your customizations */
+.phx-no-feedback.invalid-feedback,
+.phx-no-feedback .invalid-feedback {
+  display: none;
+}
+
+.phx-click-loading {
+  opacity: 0.5;
+  transition: opacity 1s ease-out;
+}
+
+.phx-disconnected{
+  cursor: wait;
+}
+.phx-disconnected *{
+  pointer-events: none;
+}
+
+.phx-modal {
+  opacity: 1!important;
+  position: fixed;
+  z-index: 1;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+  background-color: rgb(0,0,0);
+  background-color: rgba(0,0,0,0.4);
+}
+
+.phx-modal-content {
+  background-color: #fefefe;
+  margin: 15% auto;
+  padding: 20px;
+  border: 1px solid #888;
+  width: 80%;
+}
+
+.phx-modal-close {
+  color: #aaa;
+  float: right;
+  font-size: 28px;
+  font-weight: bold;
+}
+
+.phx-modal-close:hover,
+.phx-modal-close:focus {
+  color: black;
+  text-decoration: none;
+  cursor: pointer;
+}
+
+
+/* Alerts and form errors */
+.alert {
+  padding: 15px;
+  margin-bottom: 20px;
+  border: 1px solid transparent;
+  border-radius: 4px;
+}
+.alert-info {
+  color: #31708f;
+  background-color: #d9edf7;
+  border-color: #bce8f1;
+}
+.alert-warning {
+  color: #8a6d3b;
+  background-color: #fcf8e3;
+  border-color: #faebcc;
+}
+.alert-danger {
+  color: #a94442;
+  background-color: #f2dede;
+  border-color: #ebccd1;
+}
+.alert p {
+  margin-bottom: 0;
+}
+.alert:empty {
+  display: none;
+}
+.invalid-feedback {
+  color: #a94442;
+  display: block;
+  margin: -1rem 0 2rem;
+}
diff --git apps/my_app_web/assets/css/phoenix.css assets/css/phoenix.css
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/css/phoenix.css
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/css/phoenix.css
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/js/app.js assets/js/app.js
new file mode 100644
index 0000000..dfc0f97
--- /dev/null
+++ assets/js/app.js
@@ -0,0 +1,35 @@
+// We need to import the CSS so that webpack will load it.
+// The MiniCssExtractPlugin is used to separate it out into
+// its own CSS file.
+import "../css/app.scss"
+
+// webpack automatically bundles all modules in your
+// entry points. Those entry points can be configured
+// in "webpack.config.js".
+//
+// Import deps with the dep name or local files with a relative path, for example:
+//
+//     import {Socket} from "phoenix"
+//     import socket from "./socket"
+//
+import "phoenix_html"
+import {Socket} from "phoenix"
+import NProgress from "nprogress"
+import {LiveSocket} from "phoenix_live_view"
+
+let csrfToken = document.querySelector("meta[name='csrf-token']").getAttribute("content")
+let liveSocket = new LiveSocket("/live", Socket, {params: {_csrf_token: csrfToken}})
+
+// Show progress bar on live navigation and form submits
+window.addEventListener("phx:page-loading-start", info => NProgress.start())
+window.addEventListener("phx:page-loading-stop", info => NProgress.done())
+
+// connect if there are any LiveViews on the page
+liveSocket.connect()
+
+// expose liveSocket on window for web console debug logs and latency simulation:
+// >> liveSocket.enableDebug()
+// >> liveSocket.enableLatencySim(1000)  // enabled for duration of browser session
+// >> liveSocket.disableLatencySim()
+window.liveSocket = liveSocket
+
diff --git apps/my_app_web/assets/package.json assets/package.json
similarity index 79%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/package.json
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/package.json
index 624f823..aa41dee 100644
--- apps/my_app_web/assets/package.json
+++ assets/package.json
@@ -7,7 +7,10 @@
     "watch": "webpack --mode development --watch"
   },
   "dependencies": {
-    "phoenix": "file:../../../deps/phoenix"
+    "phoenix": "file:../deps/phoenix",
+    "phoenix_html": "file:../deps/phoenix_html",
+    "phoenix_live_view": "file:../deps/phoenix_live_view",
+    "nprogress": "^0.2.0"
   },
   "devDependencies": {
     "@babel/core": "^7.0.0",
diff --git apps/my_app_web/assets/static/favicon.ico assets/static/favicon.ico
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/static/favicon.ico
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/static/favicon.ico
diff --git apps/my_app_web/assets/static/images/phoenix.png assets/static/images/phoenix.png
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/static/images/phoenix.png
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/static/images/phoenix.png
diff --git apps/my_app_web/assets/static/robots.txt assets/static/robots.txt
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/static/robots.txt
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/static/robots.txt
diff --git apps/my_app_web/assets/webpack.config.js assets/webpack.config.js
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/assets/webpack.config.js
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/assets/webpack.config.js
diff --git config/config.exs config/config.exs
index b6d8a6c..4bf1199 100644
--- config/config.exs
+++ config/config.exs
@@ -1,24 +1,20 @@
-# This file is responsible for configuring your umbrella
-# and **all applications** and their dependencies with the
-# help of Mix.Config.
+# This file is responsible for configuring your application
+# and its dependencies with the aid of the Mix.Config module.
 #
-# Note that all applications in your umbrella share the
-# same configuration and dependencies, which is why they
-# all use the same configuration file. If you want different
-# configurations or dependencies per app, it is best to
-# move said applications out of the umbrella.
-use Mix.Config
-
+# This configuration file is loaded before any dependency and
+# is restricted to this project.
 
+# General application configuration
+use Mix.Config
 
-config :my_app_web,
-  generators: [context_app: :my_app, binary_id: true]
+config :my_app,
+  generators: [binary_id: true]
 
 # Configures the endpoint
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   url: [host: "localhost"],
   secret_key_base: "[redacted]",
-  render_errors: [view: MyAppWeb.ErrorView, accepts: ~w(json), layout: false],
+  render_errors: [view: MyAppWeb.ErrorView, accepts: ~w(html json), layout: false],
   pubsub_server: MyApp.PubSub,
   live_view: [signing_salt: "[redacted]"]
 
diff --git config/dev.exs config/dev.exs
index 5622e68..3459b05 100644
--- config/dev.exs
+++ config/dev.exs
@@ -6,7 +6,7 @@ use Mix.Config
 # The watchers configuration can be used to run external
 # watchers to your application. For example, we use it
 # with webpack to recompile .js and .css sources.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [port: 4000],
   debug_errors: true,
   code_reloader: true,
@@ -17,7 +17,7 @@ config :my_app_web, MyAppWeb.Endpoint,
       "--mode",
       "development",
       "--watch-stdin",
-      cd: Path.expand("../apps/my_app_web/assets", __DIR__)
+      cd: Path.expand("../assets", __DIR__)
     ]
   ]
 
@@ -45,12 +45,23 @@ config :my_app_web, MyAppWeb.Endpoint,
 # configured to run both http and https servers on
 # different ports.
 
+# Watch static and templates for browser reloading.
+config :my_app, MyAppWeb.Endpoint,
+  live_reload: [
+    patterns: [
+      ~r"priv/static/.*(js|css|png|jpeg|jpg|gif|svg)$",
+      ~r"priv/gettext/.*(po)$",
+      ~r"lib/my_app_web/(live|views)/.*(ex)$",
+      ~r"lib/my_app_web/templates/.*(eex)$"
+    ]
+  ]
+
 # Do not include metadata nor timestamps in development logs
 config :logger, :console, format: "[$level] $message\n"
 
-# Initialize plugs at runtime for faster development compilation
-config :phoenix, :plug_init_mode, :runtime
-
 # Set a higher stacktrace during development. Avoid configuring such
 # in production as building large stacktraces may be expensive.
 config :phoenix, :stacktrace_depth, 20
+
+# Initialize plugs at runtime for faster development compilation
+config :phoenix, :plug_init_mode, :runtime
diff --git config/prod.exs config/prod.exs
index 8310ab5..7443cb4 100644
--- config/prod.exs
+++ config/prod.exs
@@ -9,16 +9,19 @@ use Mix.Config
 # manifest is generated by the `mix phx.digest` task,
 # which you should run after static files are built and
 # before starting your production server.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   url: [host: "example.com", port: 80],
   cache_static_manifest: "priv/static/cache_manifest.json"
 
+# Do not print debug messages in production
+config :logger, level: :info
+
 # ## SSL Support
 #
 # To get SSL working, you will need to add the `https` key
 # to the previous section and set your `:url` port to 443:
 #
-#     config :my_app_web, MyAppWeb.Endpoint,
+#     config :my_app, MyAppWeb.Endpoint,
 #       ...
 #       url: [host: "example.com", port: 443],
 #       https: [
@@ -42,14 +45,11 @@ config :my_app_web, MyAppWeb.Endpoint,
 # We also recommend setting `force_ssl` in your endpoint, ensuring
 # no data is ever sent via http, always redirecting to https:
 #
-#     config :my_app_web, MyAppWeb.Endpoint,
+#     config :my_app, MyAppWeb.Endpoint,
 #       force_ssl: [hsts: true]
 #
 # Check `Plug.SSL` for all available options in `force_ssl`.
 
-# Do not print debug messages in production
-config :logger, level: :info
-
 # Finally import the config/prod.secret.exs which loads secrets
 # and configuration from environment variables.
 import_config "prod.secret.exs"
diff --git config/prod.secret.exs config/prod.secret.exs
index 702c128..3568fa4 100644
--- config/prod.secret.exs
+++ config/prod.secret.exs
@@ -11,7 +11,7 @@ secret_key_base =
     You can generate one by calling: mix phx.gen.secret
     """
 
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [
     port: String.to_integer(System.get_env("PORT") || "4000"),
     transport_options: [socket_opts: [:inet6]]
@@ -23,7 +23,7 @@ config :my_app_web, MyAppWeb.Endpoint,
 # If you are doing OTP releases, you need to instruct Phoenix
 # to start each relevant endpoint:
 #
-#     config :my_app_web, MyAppWeb.Endpoint, server: true
+#     config :my_app, MyAppWeb.Endpoint, server: true
 #
 # Then you can assemble a release by calling `mix release`.
 # See `mix help release` for more information.
diff --git config/test.exs config/test.exs
index 8fd529b..d363390 100644
--- config/test.exs
+++ config/test.exs
@@ -2,7 +2,7 @@ use Mix.Config
 
 # We don't run a server during test. If one is required,
 # you can enable the server option below.
-config :my_app_web, MyAppWeb.Endpoint,
+config :my_app, MyAppWeb.Endpoint,
   http: [port: 4002],
   server: false
 
diff --git apps/my_app_web/lib/my_app_web/application.ex lib/my_app/application.ex
similarity index 72%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/application.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app/application.ex
index fbdce53..9ecbdae 100644
--- apps/my_app_web/lib/my_app_web/application.ex
+++ lib/my_app/application.ex
@@ -1,4 +1,4 @@
-defmodule MyAppWeb.Application do
+defmodule MyApp.Application do
   # See https://hexdocs.pm/elixir/Application.html
   # for more information on OTP Applications
   @moduledoc false
@@ -9,15 +9,17 @@ defmodule MyAppWeb.Application do
     children = [
       # Start the Telemetry supervisor
       MyAppWeb.Telemetry,
+      # Start the PubSub system
+      {Phoenix.PubSub, name: MyApp.PubSub},
       # Start the Endpoint (http/https)
       MyAppWeb.Endpoint
-      # Start a worker by calling: MyAppWeb.Worker.start_link(arg)
-      # {MyAppWeb.Worker, arg}
+      # Start a worker by calling: MyApp.Worker.start_link(arg)
+      # {MyApp.Worker, arg}
     ]
 
     # See https://hexdocs.pm/elixir/Supervisor.html
     # for other strategies and supported options
-    opts = [strategy: :one_for_one, name: MyAppWeb.Supervisor]
+    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
     Supervisor.start_link(children, opts)
   end
 
diff --git apps/my_app/lib/my_app.ex lib/my_app.ex
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/lib/my_app.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app.ex
diff --git apps/my_app_web/lib/my_app_web/channels/user_socket.ex lib/my_app_web/channels/user_socket.ex
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/channels/user_socket.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/channels/user_socket.ex
diff --git apps/my_app_web/lib/my_app_web/endpoint.ex lib/my_app_web/endpoint.ex
similarity index 72%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/endpoint.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/endpoint.ex
index 51a8394..c620fdb 100644
--- apps/my_app_web/lib/my_app_web/endpoint.ex
+++ lib/my_app_web/endpoint.ex
@@ -1,35 +1,43 @@
 defmodule MyAppWeb.Endpoint do
-  use Phoenix.Endpoint, otp_app: :my_app_web
+  use Phoenix.Endpoint, otp_app: :my_app
 
   # The session will be stored in the cookie and signed,
   # this means its contents can be read but not tampered with.
   # Set :encryption_salt if you would also like to encrypt it.
   @session_options [
     store: :cookie,
-    key: "_my_app_web_key",
-    signing_salt: "BIDlf8VA"
+    key: "_my_app_key",
+    signing_salt: "[redacted]"
   ]
 
   socket "/socket", MyAppWeb.UserSocket,
     websocket: true,
     longpoll: false
 
+  socket "/live", Phoenix.LiveView.Socket, websocket: [connect_info: [session: @session_options]]
+
   # Serve at "/" the static files from "priv/static" directory.
   #
   # You should set gzip to true if you are running phx.digest
   # when deploying your static files in production.
   plug Plug.Static,
     at: "/",
-    from: :my_app_web,
+    from: :my_app,
     gzip: false,
     only: ~w(css fonts images js favicon.ico robots.txt)
 
   # Code reloading can be explicitly enabled under the
   # :code_reloader configuration of your endpoint.
   if code_reloading? do
+    socket "/phoenix/live_reload/socket", Phoenix.LiveReloader.Socket
+    plug Phoenix.LiveReloader
     plug Phoenix.CodeReloader
   end
 
+  plug Phoenix.LiveDashboard.RequestLogger,
+    param_key: "request_logger",
+    cookie_key: "request_logger"
+
   plug Plug.RequestId
   plug Plug.Telemetry, event_prefix: [:phoenix, :endpoint]
 
diff --git apps/my_app_web/lib/my_app_web/gettext.ex lib/my_app_web/gettext.ex
similarity index 94%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/gettext.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/gettext.ex
index 9c6644d..7919b97 100644
--- apps/my_app_web/lib/my_app_web/gettext.ex
+++ lib/my_app_web/gettext.ex
@@ -20,5 +20,5 @@ defmodule MyAppWeb.Gettext do
 
   See the [Gettext Docs](https://hexdocs.pm/gettext) for detailed usage.
   """
-  use Gettext, otp_app: :my_app_web
+  use Gettext, otp_app: :my_app
 end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/live/page_live.ex lib/my_app_web/live/page_live.ex
new file mode 100644
index 0000000..1d75b57
--- /dev/null
+++ lib/my_app_web/live/page_live.ex
@@ -0,0 +1,39 @@
+defmodule MyAppWeb.PageLive do
+  use MyAppWeb, :live_view
+
+  @impl true
+  def mount(_params, _session, socket) do
+    {:ok, assign(socket, query: "", results: %{})}
+  end
+
+  @impl true
+  def handle_event("suggest", %{"q" => query}, socket) do
+    {:noreply, assign(socket, results: search(query), query: query)}
+  end
+
+  @impl true
+  def handle_event("search", %{"q" => query}, socket) do
+    case search(query) do
+      %{^query => vsn} ->
+        {:noreply, redirect(socket, external: "https://hexdocs.pm/#{query}/#{vsn}")}
+
+      _ ->
+        {:noreply,
+         socket
+         |> put_flash(:error, "No dependencies found matching \"#{query}\"")
+         |> assign(results: %{}, query: query)}
+    end
+  end
+
+  defp search(query) do
+    if not MyAppWeb.Endpoint.config(:code_reloader) do
+      raise "action disabled when not in development"
+    end
+
+    for {app, desc, vsn} <- Application.started_applications(),
+        app = to_string(app),
+        String.starts_with?(app, query) and not List.starts_with?(desc, ~c"ERTS"),
+        into: %{},
+        do: {app, vsn}
+  end
+end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/live/page_live.html.leex lib/my_app_web/live/page_live.html.leex
new file mode 100644
index 0000000..52509c2
--- /dev/null
+++ lib/my_app_web/live/page_live.html.leex
@@ -0,0 +1,48 @@
+<section class="phx-hero">
+  <h1><%= gettext "Welcome to %{name}!", name: "Phoenix" %></h1>
+  <p>Peace of mind from prototype to production</p>
+
+  <form phx-change="suggest" phx-submit="search">
+    <input type="text" name="q" value="<%= @query %>" placeholder="Live dependency search" list="results" autocomplete="off"/>
+    <datalist id="results">
+      <%= for {app, _vsn} <- @results do %>
+        <option value="<%= app %>"><%= app %></option>
+      <% end %>
+    </datalist>
+    <button type="submit" phx-disable-with="Searching...">Go to Hexdocs</button>
+  </form>
+</section>
+
+<section class="row">
+  <article class="column">
+    <h2>Resources</h2>
+    <ul>
+      <li>
+        <a href="https://hexdocs.pm/phoenix/overview.html">Guides &amp; Docs</a>
+      </li>
+      <li>
+        <a href="https://github.com/phoenixframework/phoenix">Source</a>
+      </li>
+      <li>
+        <a href="https://github.com/phoenixframework/phoenix/blob/v1.5/CHANGELOG.md">v1.5 Changelog</a>
+      </li>
+    </ul>
+  </article>
+  <article class="column">
+    <h2>Help</h2>
+    <ul>
+      <li>
+        <a href="https://elixirforum.com/c/phoenix-forum">Forum</a>
+      </li>
+      <li>
+        <a href="https://webchat.freenode.net/?channels=elixir-lang">#elixir-lang on Freenode IRC</a>
+      </li>
+      <li>
+        <a href="https://twitter.com/elixirphoenix">Twitter @elixirphoenix</a>
+      </li>
+      <li>
+        <a href="https://elixir-slackin.herokuapp.com/">Elixir on Slack</a>
+      </li>
+    </ul>
+  </article>
+</section>
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/router.ex lib/my_app_web/router.ex
new file mode 100644
index 0000000..ad9430e
--- /dev/null
+++ lib/my_app_web/router.ex
@@ -0,0 +1,43 @@
+defmodule MyAppWeb.Router do
+  use MyAppWeb, :router
+
+  pipeline :browser do
+    plug :accepts, ["html"]
+    plug :fetch_session
+    plug :fetch_live_flash
+    plug :put_root_layout, {MyAppWeb.LayoutView, :root}
+    plug :protect_from_forgery
+    plug :put_secure_browser_headers
+  end
+
+  pipeline :api do
+    plug :accepts, ["json"]
+  end
+
+  scope "/", MyAppWeb do
+    pipe_through :browser
+
+    live "/", PageLive, :index
+  end
+
+  # Other scopes may use custom stacks.
+  # scope "/api", MyAppWeb do
+  #   pipe_through :api
+  # end
+
+  # Enables LiveDashboard only for development
+  #
+  # If you want to use the LiveDashboard in production, you should put
+  # it behind authentication and allow only admins to access it.
+  # If your application does not have an admins-only section yet,
+  # you can use Plug.BasicAuth to set up some basic authentication
+  # as long as you are also using SSL (which you should anyway).
+  if Mix.env() in [:dev, :test] do
+    import Phoenix.LiveDashboard.Router
+
+    scope "/" do
+      pipe_through :browser
+      live_dashboard "/dashboard", metrics: MyAppWeb.Telemetry
+    end
+  end
+end
diff --git apps/my_app_web/lib/my_app_web/telemetry.ex lib/my_app_web/telemetry.ex
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/telemetry.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/telemetry.ex
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/templates/layout/app.html.eex lib/my_app_web/templates/layout/app.html.eex
new file mode 100644
index 0000000..09ffdad
--- /dev/null
+++ lib/my_app_web/templates/layout/app.html.eex
@@ -0,0 +1,5 @@
+<main role="main" class="container">
+  <p class="alert alert-info" role="alert"><%= get_flash(@conn, :info) %></p>
+  <p class="alert alert-danger" role="alert"><%= get_flash(@conn, :error) %></p>
+  <%= @inner_content %>
+</main>
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/templates/layout/live.html.leex lib/my_app_web/templates/layout/live.html.leex
new file mode 100644
index 0000000..8dcded5
--- /dev/null
+++ lib/my_app_web/templates/layout/live.html.leex
@@ -0,0 +1,11 @@
+<main role="main" class="container">
+  <p class="alert alert-info" role="alert"
+    phx-click="lv:clear-flash"
+    phx-value-key="info"><%= live_flash(@flash, :info) %></p>
+
+  <p class="alert alert-danger" role="alert"
+    phx-click="lv:clear-flash"
+    phx-value-key="error"><%= live_flash(@flash, :error) %></p>
+
+  <%= @inner_content %>
+</main>
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/templates/layout/root.html.leex lib/my_app_web/templates/layout/root.html.leex
new file mode 100644
index 0000000..630a552
--- /dev/null
+++ lib/my_app_web/templates/layout/root.html.leex
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8"/>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
+    <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
+    <%= csrf_meta_tag() %>
+    <%= live_title_tag assigns[:page_title] || "MyApp", suffix: " · Phoenix Framework" %>
+    <link phx-track-static rel="stylesheet" href="<%= Routes.static_path(@conn, "/css/app.css") %>"/>
+    <script defer phx-track-static type="text/javascript" src="<%= Routes.static_path(@conn, "/js/app.js") %>"></script>
+  </head>
+  <body>
+    <header>
+      <section class="container">
+        <nav role="navigation">
+          <ul>
+            <li><a href="https://hexdocs.pm/phoenix/overview.html">Get Started</a></li>
+            <%= if function_exported?(Routes, :live_dashboard_path, 2) do %>
+              <li><%= link "LiveDashboard", to: Routes.live_dashboard_path(@conn, :home) %></li>
+            <% end %>
+          </ul>
+        </nav>
+        <a href="https://phoenixframework.org/" class="phx-logo">
+          <img src="<%= Routes.static_path(@conn, "/images/phoenix.png") %>" alt="Phoenix Framework Logo"/>
+        </a>
+      </section>
+    </header>
+    <%= @inner_content %>
+  </body>
+</html>
diff --git apps/my_app_web/lib/my_app_web/views/error_helpers.ex lib/my_app_web/views/error_helpers.ex
similarity index 78%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/views/error_helpers.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/views/error_helpers.ex
index 1882f7f..9b530af 100644
--- apps/my_app_web/lib/my_app_web/views/error_helpers.ex
+++ lib/my_app_web/views/error_helpers.ex
@@ -3,6 +3,20 @@ defmodule MyAppWeb.ErrorHelpers do
   Conveniences for translating and building error messages.
   """
 
+  use Phoenix.HTML
+
+  @doc """
+  Generates tag for inlined form input errors.
+  """
+  def error_tag(form, field) do
+    Enum.map(Keyword.get_values(form.errors, field), fn error ->
+      content_tag(:span, translate_error(error),
+        class: "invalid-feedback",
+        phx_feedback_for: input_id(form, field)
+      )
+    end)
+  end
+
   @doc """
   Translates an error message using gettext.
   """
diff --git apps/my_app_web/lib/my_app_web/views/error_view.ex lib/my_app_web/views/error_view.ex
similarity index 56%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web/views/error_view.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/views/error_view.ex
index 6bc5b14..ce4cf28 100644
--- apps/my_app_web/lib/my_app_web/views/error_view.ex
+++ lib/my_app_web/views/error_view.ex
@@ -3,14 +3,14 @@ defmodule MyAppWeb.ErrorView do
 
   # If you want to customize a particular status code
   # for a certain format, you may uncomment below.
-  # def render("500.json", _assigns) do
-  #   %{errors: %{detail: "Internal Server Error"}}
+  # def render("500.html", _assigns) do
+  #   "Internal Server Error"
   # end
 
   # By default, Phoenix returns the status message from
-  # the template name. For example, "404.json" becomes
+  # the template name. For example, "404.html" becomes
   # "Not Found".
   def template_not_found(template, _assigns) do
-    %{errors: %{detail: Phoenix.Controller.status_message_from_template(template)}}
+    Phoenix.Controller.status_message_from_template(template)
   end
 end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web/views/layout_view.ex lib/my_app_web/views/layout_view.ex
new file mode 100644
index 0000000..80fc067
--- /dev/null
+++ lib/my_app_web/views/layout_view.ex
@@ -0,0 +1,3 @@
+defmodule MyAppWeb.LayoutView do
+  use MyAppWeb, :view
+end
diff --git apps/my_app_web/lib/my_app_web.ex lib/my_app_web.ex
similarity index 78%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/lib/my_app_web.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/lib/my_app_web.ex
index 7af6bb3..222c4f0 100644
--- apps/my_app_web/lib/my_app_web.ex
+++ lib/my_app_web.ex
@@ -42,12 +42,30 @@ defmodule MyAppWeb do
     end
   end
 
+  def live_view do
+    quote do
+      use Phoenix.LiveView,
+        layout: {MyAppWeb.LayoutView, "live.html"}
+
+      unquote(view_helpers())
+    end
+  end
+
+  def live_component do
+    quote do
+      use Phoenix.LiveComponent
+
+      unquote(view_helpers())
+    end
+  end
+
   def router do
     quote do
       use Phoenix.Router
 
       import Plug.Conn
       import Phoenix.Controller
+      import Phoenix.LiveView.Router
     end
   end
 
@@ -60,6 +78,12 @@ defmodule MyAppWeb do
 
   defp view_helpers do
     quote do
+      # Use all HTML functionality (forms, tags, etc)
+      use Phoenix.HTML
+
+      # Import LiveView helpers (live_render, live_component, live_patch, etc)
+      import Phoenix.LiveView.Helpers
+
       # Import basic rendering functionality (render, render_layout, etc)
       import Phoenix.View
 
diff --git mix.exs mix.exs
index 099c509..50963ae 100644
--- mix.exs
+++ mix.exs
@@ -1,30 +1,50 @@
-defmodule MyApp.Umbrella.MixProject do
+defmodule MyApp.MixProject do
   use Mix.Project
 
   def project do
     [
-      apps_path: "apps",
+      app: :my_app,
       version: "0.1.0",
+      elixir: "~> 1.7",
+      elixirc_paths: elixirc_paths(Mix.env()),
+      compilers: [:phoenix, :gettext] ++ Mix.compilers(),
       start_permanent: Mix.env() == :prod,
-      deps: deps(),
-      aliases: aliases()
+      aliases: aliases(),
+      deps: deps()
     ]
   end
 
-  # Dependencies can be Hex packages:
+  # Configuration for the OTP application.
   #
-  #   {:mydep, "~> 0.3.0"}
-  #
-  # Or git/path repositories:
-  #
-  #   {:mydep, git: "https://github.com/elixir-lang/mydep.git", tag: "0.1.0"}
-  #
-  # Type "mix help deps" for more examples and options.
+  # Type `mix help compile.app` for more information.
+  def application do
+    [
+      mod: {MyApp.Application, []},
+      extra_applications: [:logger, :runtime_tools]
+    ]
+  end
+
+  # Specifies which paths to compile per environment.
+  defp elixirc_paths(:test), do: ["lib", "test/support"]
+  defp elixirc_paths(_), do: ["lib"]
+
+  # Specifies your project dependencies.
   #
-  # Dependencies listed here are available only for this project
-  # and cannot be accessed from applications inside the apps/ folder.
+  # Type `mix help deps` for examples and options.
   defp deps do
-    []
+    [
+      {:phoenix, "~> 1.5.7"},
+      {:phoenix_live_view, "~> 0.15.0"},
+      {:floki, ">= 0.27.0", only: :test},
+      {:phoenix_html, "~> 2.11"},
+      {:phoenix_live_reload, "~> 1.2", only: :dev},
+      {:phoenix_live_dashboard, "~> 0.4"},
+      {:telemetry_metrics, "~> 0.4"},
+      {:telemetry_poller, "~> 0.4"},
+      {:gettext, "~> 0.11"},
+      {:jason, "~> 1.0"},
+      {:plug_cowboy, "~> 2.0"}
+    ]
   end
 
   # Aliases are shortcuts or tasks specific to the current project.
@@ -33,13 +53,9 @@ defmodule MyApp.Umbrella.MixProject do
   #     $ mix setup
   #
   # See the documentation for `Mix` for more info on aliases.
-  #
-  # Aliases listed here are available only for this project
-  # and cannot be accessed from applications inside the apps/ folder.
   defp aliases do
     [
-      # run `mix setup` in all child apps
-      setup: ["cmd mix setup"]
+      setup: ["deps.get", "cmd npm install --prefix assets"]
     ]
   end
 end
diff --git apps/my_app_web/priv/gettext/en/LC_MESSAGES/errors.po priv/gettext/en/LC_MESSAGES/errors.po
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/priv/gettext/en/LC_MESSAGES/errors.po
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/priv/gettext/en/LC_MESSAGES/errors.po
diff --git apps/my_app_web/priv/gettext/errors.pot priv/gettext/errors.pot
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/priv/gettext/errors.pot
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/priv/gettext/errors.pot
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/test/my_app_web/live/page_live_test.exs test/my_app_web/live/page_live_test.exs
new file mode 100644
index 0000000..7d971b0
--- /dev/null
+++ test/my_app_web/live/page_live_test.exs
@@ -0,0 +1,11 @@
+defmodule MyAppWeb.PageLiveTest do
+  use MyAppWeb.ConnCase
+
+  import Phoenix.LiveViewTest
+
+  test "disconnected and connected render", %{conn: conn} do
+    {:ok, page_live, disconnected_html} = live(conn, "/")
+    assert disconnected_html =~ "Welcome to Phoenix!"
+    assert render(page_live) =~ "Welcome to Phoenix!"
+  end
+end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/test/my_app_web/views/error_view_test.exs test/my_app_web/views/error_view_test.exs
new file mode 100644
index 0000000..770b34b
--- /dev/null
+++ test/my_app_web/views/error_view_test.exs
@@ -0,0 +1,14 @@
+defmodule MyAppWeb.ErrorViewTest do
+  use MyAppWeb.ConnCase, async: true
+
+  # Bring render/3 and render_to_string/3 for testing custom views
+  import Phoenix.View
+
+  test "renders 404.html" do
+    assert render_to_string(MyAppWeb.ErrorView, "404.html", []) == "Not Found"
+  end
+
+  test "renders 500.html" do
+    assert render_to_string(MyAppWeb.ErrorView, "500.html", []) == "Internal Server Error"
+  end
+end
diff --git a/projects/phoenix/1.5.7/binaryid-database-noecto-live/test/my_app_web/views/layout_view_test.exs test/my_app_web/views/layout_view_test.exs
new file mode 100644
index 0000000..e1e43f4
--- /dev/null
+++ test/my_app_web/views/layout_view_test.exs
@@ -0,0 +1,8 @@
+defmodule MyAppWeb.LayoutViewTest do
+  use MyAppWeb.ConnCase, async: true
+
+  # When testing helpers, you may want to import Phoenix.HTML and
+  # use functions such as safe_to_string() to convert the helper
+  # result into an HTML string.
+  # import Phoenix.HTML
+end
diff --git apps/my_app_web/test/support/channel_case.ex test/support/channel_case.ex
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/test/support/channel_case.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/test/support/channel_case.ex
diff --git apps/my_app_web/test/support/conn_case.ex test/support/conn_case.ex
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app_web/test/support/conn_case.ex
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/test/support/conn_case.ex
diff --git apps/my_app/test/test_helper.exs test/test_helper.exs
similarity index 100%
rename from projects/phoenix/1.5.7/binaryid-noecto-nohtml-umbrella-nodash/apps/my_app/test/test_helper.exs
rename to projects/phoenix/1.5.7/binaryid-database-noecto-live/test/test_helper.exs
