diff --git a/projects/elixir/1.11.3/sup/lib/my_app/application.ex lib/my_app/application.ex
new file mode 100644
index 0000000..9de1963
--- /dev/null
+++ lib/my_app/application.ex
@@ -0,0 +1,20 @@
+defmodule MyApp.Application do
+  # See https://hexdocs.pm/elixir/Application.html
+  # for more information on OTP Applications
+  @moduledoc false
+
+  use Application
+
+  @impl true
+  def start(_type, _args) do
+    children = [
+      # Starts a worker by calling: MyApp.Worker.start_link(arg)
+      # {MyApp.Worker, arg}
+    ]
+
+    # See https://hexdocs.pm/elixir/Supervisor.html
+    # for other strategies and supported options
+    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
diff --git mix.exs mix.exs
index 169ac20..2c85fdb 100644
--- mix.exs
+++ mix.exs
@@ -14,7 +14,8 @@ defmodule MyApp.MixProject do
   # Run "mix help compile.app" to learn about applications.
   def application do
     [
-      extra_applications: [:logger]
+      extra_applications: [:logger],
+      mod: {MyApp.Application, []}
     ]
   end
 
