diff --git apps/my_app_web/lib/my_app_web/endpoint.ex apps/my_app_web/lib/my_app_web/endpoint.ex
index 72829f0..4bce966 100644
--- apps/my_app_web/lib/my_app_web/endpoint.ex
+++ apps/my_app_web/lib/my_app_web/endpoint.ex
@@ -7,7 +7,7 @@ defmodule MyAppWeb.Endpoint do
   @session_options [
     store: :cookie,
     key: "_my_app_web_key",
-    signing_salt: "lLDikEZR"
+    signing_salt: "dxIB7r7R"
   ]
 
   socket "/socket", MyAppWeb.UserSocket,
diff --git config/config.exs config/config.exs
index 9a0a3ed..cd1a522 100644
--- config/config.exs
+++ config/config.exs
@@ -15,7 +15,7 @@ config :my_app,
 
 config :my_app_web,
   ecto_repos: [MyApp.Repo],
-  generators: [context_app: :my_app, binary_id: true]
+  generators: [context_app: :my_app]
 
 # Configures the endpoint
 config :my_app_web, MyAppWeb.Endpoint,
