diff --git lib/my_app/application.ex b/projects/elixir/1.11.3/sup/lib/my_app/application.ex
deleted file mode 100644
index 9de1963..0000000
--- lib/my_app/application.ex
+++ /dev/null
@@ -1,20 +0,0 @@
-defmodule MyApp.Application do
-  # See https://hexdocs.pm/elixir/Application.html
-  # for more information on OTP Applications
-  @moduledoc false
-
-  use Application
-
-  @impl true
-  def start(_type, _args) do
-    children = [
-      # Starts a worker by calling: MyApp.Worker.start_link(arg)
-      # {MyApp.Worker, arg}
-    ]
-
-    # See https://hexdocs.pm/elixir/Supervisor.html
-    # for other strategies and supported options
-    opts = [strategy: :one_for_one, name: MyApp.Supervisor]
-    Supervisor.start_link(children, opts)
-  end
-end
diff --git mix.exs mix.exs
index 2c85fdb..169ac20 100644
--- mix.exs
+++ mix.exs
@@ -14,8 +14,7 @@ defmodule MyApp.MixProject do
   # Run "mix help compile.app" to learn about applications.
   def application do
     [
-      extra_applications: [:logger],
-      mod: {MyApp.Application, []}
+      extra_applications: [:logger]
     ]
   end
 
