diff --git apps/my_app_web/lib/my_app_web/endpoint.ex apps/my_app_web/lib/my_app_web/endpoint.ex
index 18b3a52..7c61357 100644
--- apps/my_app_web/lib/my_app_web/endpoint.ex
+++ apps/my_app_web/lib/my_app_web/endpoint.ex
@@ -7,7 +7,7 @@ defmodule MyAppWeb.Endpoint do
   @session_options [
     store: :cookie,
     key: "_my_app_web_key",
-    signing_salt: "2savEGTC"
+    signing_salt: "nllKgq+b"
   ]
 
   socket "/socket", MyAppWeb.UserSocket,
diff --git config/config.exs config/config.exs
index 173ca58..d1757ad 100644
--- config/config.exs
+++ config/config.exs
@@ -12,7 +12,7 @@ use Mix.Config
 
 
 config :my_app_web,
-  generators: [context_app: :my_app, binary_id: true]
+  generators: [context_app: :my_app]
 
 # Configures the endpoint
 config :my_app_web, MyAppWeb.Endpoint,
